packages/repository/src/__tests__/unit/model/define-model-class.unit.ts

```diff
@@ -106,4 +107,19 @@ describe('defineModelClass', () => {
     // Verify that typedefs allows us to access free-form properties
     expect(instance.name).to.equal('a name');
   });
+
+  it('should add model definition in decorator metadata', () => {
+    const definition = new ModelDefinition('Book').addProperty('title', {
+      type: 'string',
+    });
+
+    const Book = defineModelClass<typeof Model, {title: string}>(
+      Model,
+      definition,
+    );
+
+    const meta = ModelMetadataHelper.getModelMetadata(Book);
+
+    expect(meta).to.be.instanceOf(ModelDefinition);
```

Comment:
Please check that we have the expected model definition in the metadata.

```suggestion
    expect(meta).to.deepEqual(definition);
```